src/main/scala/is/hail/expr/Type.scala

```diff
@@ -191,6 +191,8 @@ sealed abstract class Type extends Serializable { self =>
   /*  Fundamental types are types that can be handled natively by RegionValueBuilder: primitive
       types, Array and Struct. */
   lazy val fundamentalType: Type = this
+
+  def containsPointer: Boolean = children.exists(_.containsPointer)
```

Comment:
This seems wrong. I think it should be `fundamentalType.containsPointer || fundamentalType.children.exists(_.containsPointer)`